servers/demo/src/routes/remote-file-drop/+page.svelte

```diff
@@ -0,0 +1,172 @@
+	let connectionStatus = 'new';
+	let connectionInfo = { link: '', qrCode: '' };
+	let progress = { currentFileNumber: 0, totalNumberOfFiles: 0, currentFileName: '' };
+	let error = '';
+
+	const copyToClipboard = (link: string) => {
+		navigator.clipboard.writeText(link);
+	};
+
+	onMount(async () => {
+		const flottformFileInputHost = new FlottformFileInputHost({
+			flottformApi: sdpExchangeServerBase,
+			createClientUrl: createCustomClientUrl
+		});
+
+		flottformFileInputHost.on('new', () => {
+			createWebRtcChannel = flottformFileInputHost.start;
+			connectionStatus = 'new';
+		});
+
+		flottformFileInputHost.on('endpoint-created', ({ link, qrCode }) => {
+			connectionInfo.link = link;
+			connectionInfo.qrCode = qrCode;
+			connectionStatus = 'endpoint-created';
+		});
+
+		flottformFileInputHost.on('connected', () => {
+			connectionStatus = 'connected';
+		});
+
+		flottformFileInputHost.on('webrtc:waiting-for-ice', () => {});
+
+		flottformFileInputHost.on('receive', () => {
+			connectionStatus = 'receive';
+		});
+
+		flottformFileInputHost.on(
+			'progress',
+			({
+				fileIndex,
+				totalFileCount,
+				fileName,
+				currentFileProgress,
+				overallProgress
+			}: {
+				fileIndex: number;
+				totalFileCount: number;
+				fileName: string;
+				currentFileProgress: number;
+				overallProgress: number;
+			}) => {
+				connectionStatus = 'progress';
+				progress.currentFileName = fileName;
+				progress.currentFileNumber = fileIndex + 1;
+				progress.totalNumberOfFiles = totalFileCount;
+			}
+		);
+
+		flottformFileInputHost.on('single-file-transferred', (receivedFile) => {
+			connectionStatus = 'single-file-transferred';
+
+			const url = window.URL.createObjectURL(receivedFile);
+			const a = document.createElement('a');
+			a.href = url;
+			a.download = receivedFile.name;
+			document.body.appendChild(a);
+			a.click();
+			document.body.removeChild(a);
+			window.URL.revokeObjectURL(url);
+		});
+
+		flottformFileInputHost.on('done', () => {
+			connectionStatus = 'done';
+		});
+
+		flottformFileInputHost.on('error', (e) => {
+			connectionStatus = 'error';
+			error = e.message;
+		});
+	});
+</script>
+
+<svelte:head>
+	<title>Flottform DEMO</title>
+</svelte:head>
+
+<div class="min-h-screen flex items-center justify-center overflow-hidden bg-gray-50">
+	<div class="max-w-screen-xl w-full p-4 box-border flex flex-col items-center">
+		<h1 class="text-2xl font-bold text-gray-800 mb-8">Remote File Drop</h1>
+
+		<div class="w-full max-w-md p-8 bg-white shadow-lg rounded-lg">
+			{#if connectionStatus === 'new'}
+				<div class="flex flex-col items-center w-full">
+					<p class="mb-8 text-center">Start a new connection to receive files</p>
+					<button
+						on:click={createWebRtcChannel}
+						class="bg-[#1a3066] hover:bg-[#2b4fa6] text-white px-4 py-2 rounded-lg text-base"
+						>Start Connection</button
+					>
+				</div>
+			{/if}
+
+			{#if connectionStatus === 'endpoint-created'}
```

Comment:
All of these could be turned into `{:else if connectionStatus === '...'}`, so it's easier to see these parts are never shown at the same time